### ISSUE TITLE
Question: varray support

### ISSUE THREAD

**Issue by wilderk** [oracle/odpi#46]
I am evaluating ODPI-C for use in our project using Master 2.1.0-dev
We have an Oracle custom data type that is a VARRAY of BINARY_DOUBLE[60].
I am following the TestFetchObjects.c example. 

Not able to retrieve the array attributes from the object. After executing the query i get the query info and object type info

```
    if (dpiStmt_getQueryInfo(stmt, 29, &queryInfo) < 0)
        return dpiSamples_showError();
    printf("('%*s', %d, %d, %d, %d, %d, %d)\n",
    		queryInfo.nameLength,
            queryInfo.name,
			queryInfo.typeInfo.oracleTypeNum,
            queryInfo.typeInfo.sizeInChars,
            queryInfo.typeInfo.clientSizeInBytes,
            queryInfo.typeInfo.precision,
			queryInfo.typeInfo.scale,
            queryInfo.nullOk);

    if (dpiObjectType_getInfo(queryInfo.typeInfo.objectType, &typeInfo) < 0)
        return dpiSamples_showError();
    printf("Fetching objects of type %.*s.%.*s\n   is collection %d\n   number of attributes %d\n",
    		typeInfo.schemaLength,
            typeInfo.schema,
			typeInfo.nameLength,
			typeInfo.name,
			typeInfo.isCollection,
			typeInfo.numAttributes);
```
And this is the output
```
('VECTOR_COVAR', 2023, 0, 0, 0, 0, 1)
Fetching objects of type DATA_OWNER.SV_VARRAY
   is collection 1
   number of attributes 0
```
Notice that the number of attributes is zero. That is not correct.  And I am unable to get values.
dpiObjectType_getAttributes()
dpiObjectAttr_getInfo()
dpiObject_getAttributeValue()

I am using Instant Client 12.2 installed locally
My development environment is Linux RHEL6

Does ODPI-C support VARRAY? If so, can you point me to an example?

**Comment by anthony-tuininga** [comment]
Yes, VARRAY is supported. This [sample](https://github.com/oracle/odpi/blob/master/samples/TestBindArrays.c) doesn't cover VARRAYs specifically but PL/SQL collections work very similarly.

The number of attributes in a collection is always zero. Instead, you need to look at this [attribute](https://oracle.github.io/odpi/doc/public_structs/dpiObjectTypeInfo.html#c.dpiObjectTypeInfo.elementTypeInfo) for the type of element, then use the functions [dpiObject_getSize()](https://oracle.github.io/odpi/doc/public_functions/dpiObject.html#c.dpiObject_getSize) and [dpiObject_getElementValueByIndex()](https://oracle.github.io/odpi/doc/public_functions/dpiObject.html#c.dpiObject_getElementValueByIndex).

That should be sufficient to get you going. If not, let me know!
